e2etest/pkg/frr/parse.go

```diff
@@ -197,9 +207,9 @@ func ParseNeighbours(vtyshRes string) ([]*Neighbor, error) {
 	res := make([]*Neighbor, 0)
 	for k, n := range toParse {
 		ip := net.ParseIP(k)
```

Comment:
we have been treating the ip in some places as the id of the neighbor in some sense, and now it would be lost (or cause mess) in unnumbered (e.g where we use `res[n.IP.String()]`) because the field would be nil, wdyt would be the right way to support it?

Reply:
Technicallly is not nil, is string `<nil>`
```
func (ip IP) String() string {
	if len(ip) == 0 {
		return "<nil>"
	}
``` which makes sure there is no panicking or anything. So if you see `<nil>` is the unnumber test.

Alternatively I could add another field `.ID` which is string and can be either `ip` or `interface` and refactor all references, but not sure is worth it.

What do you prefer? Let me know if you have a better suggestion. Thanks

Reply:
I have not been able to find a right way, we could refactor (e.g IP to be string and holds the eth0 value, but then IP is not making sense, so we would need IP to peer naming change). We could say if "<nil>" use the routeID?